ttnn/cpp/ttnn/graph/graph_trace_utils.cpp

```diff
@@ -0,0 +1,216 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "graph_trace_utils.hpp"
+
+#include "graph_processor.hpp"
+
+#include "tt_metal/common/assert.hpp"
+
+#include <unordered_set>
+#include <string>
+#include <cstdlib> // std::strtoul
+
+
+namespace ttnn::graph {
+
+namespace {
+ttnn::Shape parse_shape(const std::string_view& shape_string) {
```

Comment:
you can pass string_view buy value.

Reply:
Overall need to check if there is serialize/deserialize method for shape in future.